src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -192,31 +199,55 @@ public void BuildSymbolDefinitionMap(ISymbolNode[] definedSymbols)
             }
         }
 
-        public void EmitSymbolDefinition(int currentOffset, NodeFactory factory)
+        private string GetSymbolToEmitForTargetPlatform(string symbol)
+        {
+            string symbolToEmit = symbol;
+
+            if (_targetPlatform.OperatingSystem == TargetOS.OSX)
+            {
+                // On OSX, we need to prefix an extra underscore to account for correct linkage of 
+                // extern "C" functions.
+                symbolToEmit = "_"+symbol;
```

Comment:
Still, doesn't it feel awkward to have platform specific object writing code here? I thought the point of ObjectWriter.so/.dll was to abstract us away from the platform specific object writing rules. Should we file an issue to have this addition moved to the native library?